### ISSUE TITLE
Regression in parser for  refined givens

### ISSUE THREAD

**Issue by WojciechMazur** [scala/scala3#21958]
Regression found during work on fix for other parser regression in #21957 

## Compiler version
Last good release: 3.5.0-RC1-bin-20240506-1cdf99f-NIGHTLY
First bad release: 3.5.0-RC1-bin-20240508-b10d64e-NIGHTLY

Bisect points it could have been introduced by either ea3c688b94d9982cceda7b63969cd7e2a1887a46 or 5189e6854ad1dacc3454542c2f124f5bcb7e2a9c

## Minimized code

```Scala

trait Foo[T]:
  def foo(v: T): Unit

given myFooOfUnit: Foo[Unit]{ type X = Int } = new Foo[Unit] {
    type X = Int
    def foo(v: Unit): Unit = ???
  }

@main def Test = 
  summon[Foo[Unit]{type X = Int}]
```

## Output

```scala
[error] ./tmp.scala:5:29
[error] 'with' expected, but '{' found
[error] given myFooOfUnit: Foo[Unit]{ type X = Int } = new Foo[Unit] {
[error]                             ^
[error] ./tmp.scala:11:34
[error] '}' expected, but eof found
[error]   summon[Foo[Unit]{type X = Int}]
[error]                     
```

## Expectation
Should compile

**Comment by Gedochao** [comment]
cc @odersky @KacperFKorban

**Comment by odersky** [comment]
The syntax always specified that in an alias given the implemented type must be an `AnnotType`.

For instance, in the LTS syntax.md at https://github.com/scala/scala3/blob/lts-3.3/docs/_docs/reference/syntax.md:

```
GivenDef          ::=  [GivenSig] (AnnotType [‘=’ Expr] | StructuralInstance)
```
So I believe this was a mistake that got fixed in 3.5. The workaround is simple: Put the refined type in parens.

```scala
given myFooOfUnit: (Foo[Unit]{ type X = Int }) = ...
```

I think we can leave this as is.
